src/libNode/MicroBlockPreProcessing.cpp

```diff
@@ -664,6 +665,52 @@ void Node::PutTxnsInTempDataBase(
   }
 }
 
+void Node::SaveTxnsToS3(
+    const std::unordered_map<TxnHash, TransactionWithReceipt>&
+        processedTransactions) {
+  ostringstream oss;
+  oss << "/tmp/txns_shard_" << m_myshardId << "_txblk_"
+      << m_mediator.m_currentEpochNum;
+  string txns_filename = oss.str();
+  ofstream txns_file(txns_filename, std::fstream::binary);
+
+  for (const auto& hashTxnPair : processedTransactions) {
+    bytes serializedTxn;
+    hashTxnPair.second.Serialize(serializedTxn, 0);
+
+    // write HASH LEN and HASH
+    size_t size = hashTxnPair.first.size;
+    txns_file.write(reinterpret_cast<const char*>(&size), sizeof(size_t));
+    txns_file.write(reinterpret_cast<const char*>(hashTxnPair.first.data()),
+                    size);
+
+    // write TXN LEN AND TXN
+    size = serializedTxn.size();
+    txns_file.write(reinterpret_cast<const char*>(&size), sizeof(size_t));
+    txns_file.write(reinterpret_cast<const char*>(serializedTxn.data()), size);
+  }
+
+  txns_file.close();
+
+  // upload to S3
+  if (!SysCommand::ExecuteCmd(SysCommand::WITHOUT_OUTPUT,
+                              GetAwsS3CpString(txns_filename))) {
```

Comment:
Do we want to keep the `/tmp/` part of the filename when we upload to S3?

Reply:
/tmp/ part is used by `aws s3 cp` to only get the local path of file/object. Only the leaf object without `/tmp` part is copied over to S3.